docs/docs/multipart.md

```diff
@@ -0,0 +1,261 @@
+# Multipart and Form Handling
+
+Multipart is a content type for messages (requests or responses) composed of multiple parts.
+Each part is itself similar to a message in that it has its own body and headers. A common use
+for multipart requests is for user-submitted forms, especially ones that include files.
+Browsers also support (and default to) another encoding for forms, `application/x-www-form-urlencoded`.
+This encoding is simpler, but is not often used for binary data.
+
+## UrlForm
+
+To handle `application/x-www-form-urlencoded` messages, http4s provides `UrlForm` and respective `EntityEncoder` and 
+`EntityDecoder`. The following example shows the client sending a form request and a server parsing it:
+
+```scala mdoc:silent
+import org.http4s.client.Client
+import cats.effect._
+import org.http4s._
+import org.http4s.dsl.io._
+import org.http4s.headers._
+import org.http4s.multipart._
+import org.http4s.implicits._
+
+import cats.effect.unsafe.implicits.global
+
+val routes = HttpRoutes
+  .of[IO] { case request @ POST -> Root / "url-form" =>
+    request.as[UrlForm].flatMap { form =>
+      val name = form.values
+        .collectFirst { case ("name", values) => values }
+        .flatMap(_.headOption)
+        .getOrElse("")
+      Ok(s"Hello, $name")
+    }
+  }
+
+val client = Client.fromHttpApp(routes.orNotFound)
+val request = Request[IO](
+  method = POST,
+  uri = uri"http://example/url-form",
+).withEntity(UrlForm("name" -> "Duncan", "version" -> "4"))
+
+```
+```scala mdoc
+client.expect[String](request)
+  .unsafeRunSync()
+```
+
+You can try this self-contained example using `scala-cli` and pointing your browser to http://localhost:8089/.
```

Comment:
`scala-cli` is cool but... couldn't it be yet another stair to overstep for newcomers?

Reply:
scala-cli is very popular with newcomers. It's probably the first stair they've stepped :)

Anyway, a reader is totally free to ignore that.

Reply:
I wouldn't be so confident (given my own experience with scala-cli), but if you say that, then I have nothing to set against 👍🏻

Reply:
I think the use of scala-cli is good here, but let's have the mention of `scala-cli` be a link to their webpage.